palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/model/AuditablePalisadeSystemResponse.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.palisade.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import uk.gov.gchq.palisade.Generated;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class is a container for {@code PalisadeClientRequest} and {@code AuditErrorMessage} during stream processing.
+ * Under normal conditions only one of these will be non-null, indicating failed or successful processing.
+ */
+
+public final class AuditablePalisadeSystemResponse {
+
+    private final PalisadeSystemResponse palisadeResponse;
+    private final AuditErrorMessage auditErrorMessage;
+
+    @JsonCreator
+    private AuditablePalisadeSystemResponse(
+            final PalisadeSystemResponse palisadeResponse,
+            final AuditErrorMessage auditErrorMessage) {
+        this.palisadeResponse = palisadeResponse;
+        this.auditErrorMessage = auditErrorMessage;
+    }
+
+    /**
+     * Chain any errors from previous stream elements
+     *
+     * @param audit the previous audit or null
+     * @return a new instance of this object
+     */
+    public AuditablePalisadeSystemResponse chain(final AuditErrorMessage audit) {
+        return Optional.ofNullable(audit).map(message -> Builder.create()
+                .withAuditErrorMessage(message))
+                .orElse(this);
+    }
+
+    @Generated
+    public PalisadeSystemResponse getPalisadeResponse() {
+        return palisadeResponse;
+    }
+
+    @Generated
+    public AuditErrorMessage getAuditErrorMessage() {
+        return auditErrorMessage;
+    }
+
+    /**
+     * The static builder
+     */
+    public static class Builder {
+
+        /**
+         * The creator function
+         *
+         * @return the composed immutable object
+         */
+        public static IPalisadeRequest create() {
+            return AuditablePalisadeSystemResponse::new;
+        }
+
+        /**
+         * Compose with {@code UserResponse}
+         */
+        public interface IPalisadeRequest {
```

Comment:
Oops, pretty sure there is no `UserResponse` here.
```suggestion
        /**
         * Compose with {@code PalisadeSystemResponse}
         */
        public interface IPalisadeRequest {
```